### ISSUE TITLE
Why classifier fine-tuning don't save best model based on the evaluation on dev dataset

### ISSUE THREAD

**Issue by nghuyong** [huggingface/pytorch-pretrained-BERT#529]
I want to use bert to train a classify model, I use the example [run_classifier.py].
But I find that the model will continue to train on train dataset until the max_epoch, without doing the evaluation on the dev dataset and save the best model according to the metric on the dev dataset.
So, the final saved model, it just the last epoch, but this saved model will not be the best model on the dev dataset!
Also, I suggest adding a args --predict and only make predictions.
This work helps me a lot! Thanks!

**Comment by Nidhi-kumari** [comment]
were you able to fix this problem. If yes can you please tell how
